common/djangoapps/student/views.py

```diff
@@ -1962,6 +1970,9 @@ def activate_account(request, key):
                             manual_enrollment_audit.reason, enrollment
                         )
 
+        if not already_active:
+            return redirect('/welcome')
+
```

Comment:
Same as second comment above. Use `settings.MKTG_URLS['WELCOME']` to verify if the URL is set before redirecting.